_posts/2025-05-23-5.12.0-released.md

```diff
@@ -0,0 +1,113 @@
+    <video autoplay controls loop>
+        <source src="/static/blog/5.12.0/animated_inventory_nodes.webm" type="video/webm" />
+    </video>
+    <figcaption>Animated nodes in the inventory</figcaption>
+</figure>
+
+Texture filtering is [no longer a blurry mess](https://github.com/luanti-org/luanti/pull/16034), softening pixels as intended. Gone are the days of screenshots ruined by bilinear and trilinear filtering!
+
+{% include figure.html src="/static/blog/5.12.0/trilinear_filter.webp" caption="Trilinear filtering before and after (note corners on flowers)" %}
+
+Package pages in the content browser [now have a _Reviews_ tab](https://github.com/luanti-org/luanti/pull/15254) where you can see exactly what other users like (or dislike) about that thing you want to download. Also pictured below is the [new default flat button style](https://github.com/luanti-org/luanti/pull/16015) with the 2000s-style gradient removed.
+
+{% include figure.html src="/static/blog/5.12.0/content_reviews_tab.webp" caption="These reviews carry a lot of weight!" %}
+
+## Camera Mode, Entity Visuals, and More
+
+Are you making a game with an exclusively 3rd-person camera? Need the player to take a good look at themselves? The [new `set_camera` player method](https://github.com/luanti-org/luanti/pull/15796) allows developers to set the current camera mode by force! This method will hopefully expand in the future.
+
+The `node` visual type [has been introduced](https://github.com/luanti-org/luanti/pull/15683) to entities to represent any node (instead of hacking the `item` visual til it looks right).
+
+```lua
+initial_properties = {
+    visual = "node",
+    node = {
+        name = "mymod:mynode",
+        param2 = 42, -- It supports paramtypes too!
+    }
+}
+```
+
+Developers can now prevent the `Escape` key from closing their formspecs using [the `allow_close[true]` element](https://github.com/luanti-org/luanti/pull/15971). The form can still be closed with an exit button or with `Shift + Esc` in emergencies (so don't rely on your forms being locked open). The `try_quit` event field has also been introduced to watch for close attempts on locked forms.
+
+As for notable fixes, custom hands now [properly override all tool capabilities](https://github.com/luanti-org/luanti/pull/15743). glTF models saw various fixes, most importantly local transforms [no longer affect meshes](https://github.com/luanti-org/luanti/pull/16086).
+
+## World and Object Performance
+
+Alongside better world [loading](https://github.com/luanti-org/luanti/pull/16000) and [generation](https://github.com/luanti-org/luanti/pull/15856) performance, world storage [has been changed](https://github.com/luanti-org/luanti/pull/15768) to save block positions as separate integers instead of a hash. This simpler implementation is much easier to work with internally and for external tools. **Please note:** Worlds created with or migrated to 5.12.0 are **not backward-compatible.**
```

Comment:
```suggestion
Alongside better world [loading](https://github.com/luanti-org/luanti/pull/16000) and [generation](https://github.com/luanti-org/luanti/pull/15856) performance, world storage [has been changed](https://github.com/luanti-org/luanti/pull/15768) to save block positions as separate integers instead of a hash. This simpler implementation is much easier to work with internally and for external tools. **Please note:** Worlds created with or migrated to 5.12.0 are cannot be opened in older versions.
```

Reply:
Nevermind this is not correct https://forum.luanti.org/viewtopic.php?p=444875#p444875